### DISCUSSION TITLE
Zustand caching issue with Next.js 13.4 SSR and server components.

### DISCUSSION THREAD

**Discussion by haramishra** [pmndrs/zustand#2058] [Bug report] [answered]
https://github.com/pmndrs/zustand/assets/28607633/86658a8f-ac3f-4a9c-ae5a-d1b077c2396a

I implemented Zustand using [this guide](https://www.youtube.com/watch?v=OpMAH2hzKi8), but when multiple users visit the website, the store is shared between the users.

The above video shows the billing and pricing component that should render differently for authenticated and unauthenticated users. However, states are shared between the two users. I added "force-dynamic" to the pages, but the issue remains the same.

Repo - https://github.com/haramishra/supanext-nextjs13-starter/tree/with-zustand
page.tsx - https://github.com/haramishra/supanext-nextjs13-starter/blob/with-zustand/src/app/page.tsx
layout.tsx - https://github.com/haramishra/supanext-nextjs13-starter/blob/with-zustand/src/app/layout.tsx

**Comment by dai-shi** [comment, marked as answer]
Sharing state across sessions in SSR is a known behavior.
You want to use React Context to avoid it.
